src/main/java/Revers.java

```diff
@@ -0,0 +1,31 @@
+import java.util.*;
+
+public class Revers {
+
+    public static <K, V> Map<K, V> createMapOption1(List<K> keys, List<V> values) {
+        Map<K, V> map = new HashMap<>();
+        for (K key : keys) {
+            if (values.isEmpty()) {
+                map.put(key, null);
+            } else {
+                map.put(key, values.remove(0));
+            }
+        }
+        return map;
+    }
+
+    public static <K, V> Map<K, V> createMapOption2(List<K> keys, List<V> values) throws Exception {
+        Map<K, V> map = new HashMap<>();
+        if (keys.size() != values.size()) {
+            String exceptionMessage = keys.size() > values.size() ?
+                    "Value is not enough for the Map" :
+                    "Key is not enough for the Map";
+            throw new Exception(exceptionMessage);
+        }
+        for (K k : keys) {
+            int index = keys.indexOf(k);
```

Comment:
you can get rid of this variable and use result of the indexOf() on the next line: `map.put(k, values.get(keys.indexOf(k)));`